rustls/src/msgs/message.rs

```diff
@@ -406,26 +414,48 @@ impl TryFrom<PlainMessage> for Message<'static> {
 ///
 /// A [`PlainMessage`] must contain plaintext content. Encrypted content should be stored in an
 /// [`OpaqueMessage`] and decrypted before being stored into a [`PlainMessage`].
-impl<'a> TryFrom<BorrowedPlainMessage<'a>> for Message<'a> {
+impl<'a> TryFrom<BorrowedPlainMessageInbound<'a>> for Message<'a> {
     type Error = Error;
 
-    fn try_from(plain: BorrowedPlainMessage<'a>) -> Result<Self, Self::Error> {
-        let payload = match plain.payload {
-            BorrowedPlainPayload::Empty => &[],
-            BorrowedPlainPayload::Single(chunk) => chunk,
-            BorrowedPlainPayload::Multiple { .. } => {
-                // for now the try_from is only used in first_handshake_message which gets its
-                // BorrowedPlainMessage directly from the deframer, so this branch is unreachabe
-                unimplemented!()
-            }
-        };
+    fn try_from(plain: BorrowedPlainMessageInbound<'a>) -> Result<Self, Self::Error> {
         Ok(Self {
             version: plain.version,
-            payload: MessagePayload::new(plain.typ, plain.version, payload)?,
+            payload: MessagePayload::new(plain.typ, plain.version, plain.payload)?,
         })
     }
 }
 
+pub trait BorrowedPlainMessage: Sized {
+    fn typ(&self) -> ContentType;
+    fn version(&self) -> ProtocolVersion;
+
+    fn payload_to_vec(&self) -> Vec<u8>;
+
+    fn payload_len(&self) -> usize;
+    fn encoded_len(&self, record_layer: &RecordLayer) -> usize {
+        OpaqueMessage::HEADER_SIZE as usize + record_layer.encrypted_len(self.payload_len())
+    }
```

Comment:
nit: I think this would look better with consistent spacing between the trait fns at the top, 1 line of whitespace each.